token/services/tokens/storage.go

```diff
@@ -114,12 +113,20 @@ func (t *transaction) AppendToken(
 	if err != nil {
 		return errors.Wrapf(err, "cannot covert [%s] with precision [%d]", tok.Quantity, precision)
 	}
+	id, err := identity.UnmarshalTypedIdentity(tok.Owner.Raw)
```

Comment:
I'm wondering: Should we replace this with an `unmarshaller` that given in input the raw []bytes gives the type and the identity? 
It is just to detach the tokens package from the identity package.
In theory, the content of an identity is driver dependant, but we can make in a future PR explicit the presence of the type.

Reply:
I like the idea of decoupling them on this level, but where should the unmarshaler be? Let me know if you have an example that I can reuse.